src/internal/storage/metrics/metrics.go

```diff
@@ -81,14 +89,21 @@ type callInfo struct {
 	line        int
 }
 
-func retrieveCallInfo(skip ...int) *callInfo {
+func retrieveCallInfo(skip ...int) (*callInfo, error) {
```

Comment:
BTW, another potential refactoring here is to just make retrieveCallInfo take the name of the function that's calling it, and then upstream you end up with something like `MonitorThroughput("GetFoo", func () ...)` instead 😂   You don't have to fix that here, I like what you have.